[quant][be] Move QAT tests to its own file

andrewor14 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #108061

Test Plan:
python test/test_quantization.py TestQuantizePT2EQAT
python test/test_quantization.py TestQuantizePT2EQATModels

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

andrewor14 committed:
[quant][be] Move QAT tests to its own class

Test Plan:
python test/test_quantization.py TestQuantizePT2EQAT

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

[ghstack-poisoned]

jerryzh168 reviewed (approved):
thanks!

andrewor14 committed:
Update on "[quant][be] Move QAT tests to its own class"


Test Plan:
python test/test_quantization.py TestQuantizePT2EQAT

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

[ghstack-poisoned]

jerryzh168 commented on test/quantization/pt2e/test_quantize_pt2e_qat.py:
```diff
@@ -0,0 +1,489 @@
+from torch.testing._internal.common_quantized import override_quantized_engine
+
+# TODO: merge all TestHelperModules and put them in a shared utils file
+from .test_quantize_pt2e import TestHelperModules
```
if you rebase on main branch this is moved to common_quantization.py

andrewor14 committed:
Update on "[quant][be] Move QAT tests to its own file"


Test Plan:
python test/test_quantization.py TestQuantizePT2EQAT
python test/test_quantization.py TestQuantizePT2EQATModels

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

[ghstack-poisoned]

andrewor14 committed:
Update on "[quant][be] Move QAT tests to its own file"


Test Plan:
python test/test_quantization.py TestQuantizePT2EQAT
python test/test_quantization.py TestQuantizePT2EQATModels

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

[ghstack-poisoned]

andrewor14 commented:
@pytorchbot merge
